src/Data/Unrestricted/Linear/Internal/Dupable.hs

```diff
@@ -1,28 +1,50 @@
+{- options_ghc -ddump-simpl -dsuppress-type-applications -dsuppress-coercions #-}
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE PolyKinds #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# OPTIONS_HADDOCK hide #-}
 
 module Data.Unrestricted.Linear.Internal.Dupable
   ( Dupable (..),
+    genericdupR,
     dup,
     dup3,
     dup4,
     dup5,
     dup6,
     dup7,
+    GDupable,
   )
 where
 
+import Data.List.NonEmpty (NonEmpty)
 import Data.Replicator.Linear.Internal (Replicator (..))
 import qualified Data.Replicator.Linear.Internal as Replicator
 import Data.Replicator.Linear.Internal.ReplicationStream (ReplicationStream (..))
+import qualified Data.Replicator.Linear.Internal.ReplicationStream as ReplicationStream
+import qualified Data.Semigroup as Semigroup
 import Data.Unrestricted.Linear.Internal.Consumable
+import Data.Unrestricted.Linear.Internal.Ur (Ur)
+import GHC.Tuple (Solo (..))
+import GHC.Types (Multiplicity (..))
+import Generics.Linear
+import Prelude.Linear.Generically
 import Prelude.Linear.Internal
+import qualified Unsafe.Linear as Unsafe
+import qualified Prelude
 
 -- | The laws of 'Dupable' are dual to those of 'Monoid':
 --
```

Comment:
>
```
-- Implementation of 'Dupable' for 'Data.Unrestricted.Movable' types should
-- be done with @deriving via 'Data.Unrestricted.AsMovable'@.
```

We should add a mention about generic deriving in the docstring of `Dupable` IMO.

Reply:
Good idea.